{{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/resource.py

```diff
@@ -1,33 +1,56 @@
-# pylint: disable=W0511, W0613
-"""
-Demo resource strategy class
-"""
+"""Demo resource strategy class."""
+# pylint: disable=no-self-use,unused-argument
 from dataclasses import dataclass
-from typing import Any, Dict, Optional
+from typing import TYPE_CHECKING
 
-from oteapi.models.resourceconfig import ResourceConfig
-from oteapi.plugins.factories import StrategyFactory
-from oteapi.plugins.factories import create_download_strategy
+from oteapi.plugins.factories import StrategyFactory, create_download_strategy
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Optional
+
+    from oteapi.models.resourceconfig import ResourceConfig
 
 
 @dataclass
 @StrategyFactory.register(("accessService", "DEMO-access-service"))
 class DemoResourceStrategy:
-    """Resource Interface"""
+    """Resource Strategy."""
+
+    resource_config: "ResourceConfig"
+
+    def initialize(
+        self, session: "Optional[Dict[str, Any]]" = None
+    ) -> "Dict[str, Any]":
+        """Initialize strategy.
+
+        This method will be called through the `/initialize` endpoint of the OTE-API
+        Services.
 
-    resource_config: ResourceConfig
+        Parameters:
+            session: A session-specific dictionary context.
 
-    def initialize(self, session: Optional[Dict[str, Any]] = None) -> Dict:
-        """Initialize"""
+        Returns:
+            Dictionary of key/value-pairs to be stored in the sessions-specific
+            dictionary context.
 
+        """
         return {}
 
-    def get(self, session: Optional[Dict[str, Any]] = None) -> Dict:
-        """Manage mapping and return shared map"""
+    def get(self, session: "Optional[Dict[str, Any]]" = None) -> "Dict[str, Any]":
+        """Execute the strategy.
 
+        This method will be called through the strategy-specific endpoint of the
+        OTE-API Services.
+
+        Parameters:
+            session: A session-specific dictionary context.
+
+        Returns:
+            Dictionary of key/value-pairs to be stored in the sessions-specific
+            dictionary context.
+
+        """
         # Example of the plugin using the download strategy to fetch the data
         download_strategy = create_download_strategy(self.resource_config)
-        read_output = download_strategy.read({})
-        print(read_output)
-
-        return {}
+        read_output = download_strategy.get(session)
```

Comment:
This might be a stupid question, but is it necessary to use the session here? Looking at the get() functions of the download strategies in oteapi-core, I can't see that any of them use the session for anything.

Reply:
The session is just passed on to the download strategy in case it needs it. The session may e.g. be used to pass configurations from a filter's initialise() to a download strategy's initialise(). Not sure about the exact use case for passing the session to get(), but I think there might be similar cases for that.

Reply:
And for them not using the `session`, this will differ wildly from strategy to strategy.
I like the idea of some standard practices and usages of `session` however that we can think about and implement - like ensuring the local variables are always updated according to the values in the `session` - before falling back on default values, always check the `session` for that given value - etc.

Reply:
> They all take in the session dictionary, which is specific to each call and is utilized to its full intent through otelib and the pipeline system. If you are familiar with "context"-concepts in programming this is essentially that. It's a set up variables that is shared and passed along as the "context" for a single "run" through the uses of a set of strategies :)

I am not familiar with "context"-concepts in programming, but I will try to read up on it + look closer at otelib to try and make sense of this.. I think I understand the point of passing the session along, but don't understand _how_ that really happens when the functions don't use or return the session. But oh well, the PR looks good to me now and I will approve it! :)

Reply:
Let's chat about it over a rundown of otelib at some point :)